pallets/msa/src/lib.rs

```diff
@@ -101,6 +139,26 @@ impl<T: Config> Pallet<T> {
 			Ok(())
 		})
 	}
+
+	pub fn is_msa_owner(who: &T::AccountId, msa_id: MessageSenderId) -> DispatchResult {
+		ensure!(Self::get_owner_of(&who) == Some(msa_id), Error::<T>::NotMsaOwner);
+		Ok(())
+	}
+
+	pub fn verify_signature(
+		signature: MultiSignature,
+		signer: T::AccountId,
+		payload: Vec<u8>,
+	) -> DispatchResult {
+		let key = T::ConvertIntoAccountId32::convert(signer.clone());
+		let wrapped_payload = wrap_binary_data(payload);
```

Comment:
so we assume all the payloads should be wrapped for now

Reply:
Yes.